packages/repository/src/query.ts

```diff
@@ -443,31 +463,36 @@ export class FilterBuilder {
   }
 
   /**
-   * Add a filter object. For conflicting keys with its where object,
-   * create an `and` clause. For any other properties, throw an error.
-   * @param filter filter object
+   * Add an object with open properties. If it is a filter object, create an `and`
+   * clause for conflicting keys with its where object. For any other properties,
+   * throw an error.
+   * @param constraint a constraint object to merge with own filter object
    */
-  impose(filter: Filter): this {
+  impose(constraint: Filter | Where): this {
```

Comment:
> if not a filter, then treat it as a where object). Thoughts?

I am little bit worried about the confusion this dual type (Filter | Where) can cause between users - I personally would not be sure which form to use when writing the code and whether a correct form is used when reading code.

Having said that, I don't have any good arguments to support my concerns, and I do see value in allowing both Filter and Where objects. I'll leave the ultimate decision on this to somebody else to make.

Reply:
I see your concern @bajtos, but yea it's to provide value so that when users provide a `Where` object for a constraint, we coerce that to a filter and provide convenience. We will document it in the TSDocs and if it confuses users like you said, we can be removed later.